src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoRequestMapper;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.mapper.impl.request.ShoppingCartRequestMapper;
+import mate.academy.spring.mapper.impl.response.OrderResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.dto.request.ShoppingCartRequestDto;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderResponseMapper;
+    private final DtoRequestMapper<ShoppingCartRequestDto, ShoppingCart> shoppingCartRequestMapper;
+    private final UserService userService;
+
+    public OrderController(OrderService orderService,
+                           OrderResponseMapper orderResponseMapper,
+                           ShoppingCartRequestMapper shoppingCartRequestMapper,
+                           UserService userService) {
+        this.orderService = orderService;
+        this.orderResponseMapper = orderResponseMapper;
+        this.shoppingCartRequestMapper = shoppingCartRequestMapper;
+        this.userService = userService;
+    }
+
+    @PostMapping("/{id}")
+    OrderResponseDto completeOrder(@PathVariable Long id,
+                                   @RequestBody ShoppingCartRequestDto shoppingCartRequestDto) {
```

Comment:
```suggestion
    OrderResponseDto completeOrder(@PathVariable Long id) {
```
you dont need to pass dto, all the information that you need is already in db